.github/workflows/ci.yml

```diff
@@ -13,37 +13,40 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        cabal: ["3.2"]
+        cabal: ["3.4"]
         ghc:
-          - "8.10.1"
-          - "8.10.2"
+          - "9.0.0.20201227"
 
     steps:
     - uses: actions/checkout@v2
-      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/ghc-8.10'
+      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/ghc-9.0'
 
-    - uses: actions/setup-haskell@v1.1.4
+    - uses: haskell/actions/setup@v1
       id: setup-haskell-cabal
       name: Setup Haskell
       with:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
 
-    - name: Freeze
+    - name: Prepare environment
       run: |
-        cabal freeze
+        echo "$HOME/.ghcup/bin" >> $GITHUB_PATH
+        sudo apt-get install libtinfo-dev
 
-    - uses: actions/cache@v1
+    - name: Freeze
+      run: |
+        cabal freeze --allow-newer
```

Comment:
It’s probably a better idea to update `cabal.project` accordingly so that local builds work too, see #1305 1305